src/javascript/app/pages/cashier/payment_agent_list.js

```diff
@@ -16,6 +16,11 @@ const PaymentAgentList = (() => {
             });
         });
 
+        if (!Client.get('currency')) {
```

Comment:
same here, why do we go through doing the accordion action before redirecting? it should be the first thing we do